Sources/ConsoleKit/Utilities/ConsoleLogger.swift

```diff
@@ -46,66 +152,35 @@ public struct ConsoleLogger: LogHandler {
         set { self.metadata[key] = newValue }
     }
     
-    /// See `LogHandler.log(level:message:metadata:file:function:line:)`.
+    /// See `LogHandler.log(level:message:metadata:source:file:function:line:)`.
     public func log(
         level: Logger.Level,
         message: Logger.Message,
         metadata: Logger.Metadata?,
+		source: String,
         file: String,
         function: String,
         line: UInt
     ) {
-        var text: ConsoleText = ""
-        
-        if self.logLevel <= .trace {
-            text += "[ \(self.label) ] ".consoleText()
-        }
-            
-        text += "[ \(level.name) ]".consoleText(level.style)
-            + " "
-            + message.description.consoleText()
-
-        let allMetadata = (metadata ?? [:])
-            .merging(self.metadata, uniquingKeysWith: { (a, _) in a })
-            .merging(self.metadataProvider?.get() ?? [:], uniquingKeysWith: { (a, _) in a })
-
-        if !allMetadata.isEmpty {
-            // only log metadata if not empty
-            text += " " + allMetadata.sortedDescriptionWithoutQuotes.consoleText()
-        }
-
-        // log file info if we are debug or lower
-        if self.logLevel <= .debug {
-            // log the concise path + line
-            let fileInfo = self.conciseSourcePath(file) + ":" + line.description
-            text += " (" + fileInfo.consoleText() + ")"
-        }
+        var output = FragmentOutput()
 
-        self.console.output(text)
-    }
-    
-    /// splits a path on the /Sources/ folder, returning everything after
-    ///
-    ///     "/Users/developer/dev/MyApp/Sources/Run/main.swift"
-    ///     // becomes
-    ///     "Run/main.swift"
-    ///
-    private func conciseSourcePath(_ path: String) -> String {
-        let separator: Substring = path.contains("Sources") ? "Sources" : "Tests"
-        return path.split(separator: "/")
-            .split(separator: separator)
-            .last?
-            .joined(separator: "/") ?? path
-    }
```

Comment:
I'm willing to bet the lack of this is what's making this branch faster. I'm realizing as I look at this logic that it's not exactly a trivial code path. `split()` in particular is both allocating and `O(n log n)` at _best_ (and can get as bad as `O(2^n)` in a pinch) - the fact that `#fileID` means we don't have to do it at all is probably a large win.

Reply:
Ahh yep, I tried running the test with just the path and that gets averages of about 0.45 seconds. So main without the file path manipulation is a hair faster than the logger-fragment version, but the difference is relatively small. About a microsecond per logged message on my machine, I think?